samples/go/downloader/treewalker.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"github.com/attic-labs/noms/go/d"
+	"github.com/attic-labs/noms/go/hash"
+	"github.com/attic-labs/noms/go/types"
+)
+
+// TreeWalkCallback defines a function prototype submitting successive values
+// from TreeWalk to the caller. At any point, the function can return true
+// to prevent traversing any of the current value's children.
+type TreeWalkCallback func(p types.Path, parent, v types.Value) (stop bool)
+
+// TreeWalk walks over a noms graph starting at 'v' and calls 'twcb' for each
```

Comment:
Should we just update the built-in noms walk to maintain the path and pass it to callee?